### PR TITLE
Added Start Flow Comms (AC -> EReg)

### PR DESCRIPTION
**Pull request by curtisjhu**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,6 +6,15 @@
         "unordered_map": "cpp",
         "vector": "cpp",
         "string_view": "cpp",
-        "initializer_list": "cpp"
+        "initializer_list": "cpp",
```
Comment (minhn02) [inline .vscode/settings.json:9]: don't add these files when they pop up; should only commit files you directly change, not config files

File: src/AC/AC1/EReg.cpp [hunk 2]
```diff
@@ -193,14 +197,13 @@ namespace EReg {
...
+        for(ERegBoard &board : *eregBoards)
+            Comms::emitPacket(&eregStartFlowPacket, (uint8_t) board->ip_address);
+        Comms::emitPacket(&eregStartFlowPacket, 22);
     }
 
     void abort() {
-        //TODO implement individual aborts?
-        fuelBoard.sendSerial(&eregAbortPacket);
-        loxBoard.sendSerial(&eregAbortPacket);
+        Comms::emitPacket(&eregStartFlowPacket, (uint8_t) board->ip_address);
```
Comment (minhn02) [inline src/AC/AC1/EReg.cpp]: should send abort packet. send to eregdaq too

File: src/AC/AC2/LoadCells.cpp [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace LoadCells {
     float loadCellSum;
     float lastLoadCellTime;
 
-    Comms::Packet lcAbortPacket = {.id = 31};
+    Comms::Packet abortPacket = {.id = 1};
```
Comment (minhn02) [inline src/AC/AC2/LoadCells.cpp]: abort should be 201

File: src/AC/AC2/LoadCells.cpp [hunk 4]
```diff
@@ -33,16 +33,22 @@ namespace LoadCells {
         return samplePeriod;
     }
 
-    uint32_t checkForLCAbort() {
+    uint32_t checkForAbort() {
```
Comment (minhn02) [inline src/AC/AC2/LoadCells.cpp]: call it LCAbort

File: src/AC/AC2/Thermocouples.cpp [hunk 5]
```diff
@@ -6,7 +6,7 @@ namespace Thermocouples {
 
     uint8_t hysteresisValues[5] = {0};
 
-    Comms::Packet tcAbortPacket = {.id = 30};
+    Comms::Packet abortPacket = {.id = 1};
```
Comment (minhn02) [inline src/AC/AC2/Thermocouples.cpp]: should change to same id as above 201

File: src/AC/AC2/Thermocouples.cpp [hunk 6]
```diff
@@ -116,4 +116,9 @@ namespace Thermocouples {
         return tcUpdatePeriod;
     }
 
+    bool abortAll() {
```
Comment (minhn02) [inline src/AC/AC2/Thermocouples.cpp:103]: you should reference this ip's in the HAL or a config file so you don't have to change them everytime you change an ip

File: src/AC/ERegBoard.cpp [hunk 7]
```diff
@@ -16,43 +16,8 @@ ERegBoard::ERegBoard(HardwareSerial *serial, uint8_t id) {
...
-                packetBufferCtr_++;
-            }
-            Comms::Packet *packet = (Comms::Packet *)packetBuffer_;
-            // DEBUG("Got unverified packet with ID ");
-            // DEBUG(packet->id);
-            // DEBUG('\n');
-            return packet;
-    }
-    return &failPacket_;
+void ERegBoard::sendEthernet(Comms::Packet *packet, uint8_t ip) {
```
Comment (minhn02) [inline src/AC/ERegBoard.cpp:19]: you have to add the checksum and timestamp before sending the packet, see emitPacket in comms for how to do that

File: src/EReg/StateMachine.cpp [hunk 8]
```diff
@@ -91,6 +91,16 @@ namespace StateMachine {
     void checkAbortPressure(float currentPressure, float abortPressure) {
         if (currentPressure > abortPressure) {
             Packets::sendFlowState(0);
+
+            Comms::Packets abortPacket = { .id = 51 };
```
Comment (minhn02) [inline src/EReg/StateMachine.cpp]: probably should just have a universal abort packet with a code field

File: src/EReg/StateMachine.cpp [hunk 9]
```diff
@@ -91,6 +91,16 @@ namespace StateMachine {
...
+            Comms::emitPacket(&abortPacket, 21);
+
+            Comms::Packet eregAbortPacket = { .id = 1 };
```
Comment (minhn02) [inline src/EReg/StateMachine.cpp]: same thing

File: src/AC/AC1/EReg.cpp [hunk 10]
```diff
@@ -34,10 +37,12 @@ namespace EReg {
     Comms::Packet fuelFlowStatePacket = {.id = 16};
     Comms::Packet loxFlowStatePacket = {.id = 17};
 
-    ERegBoard fuelBoard(&Serial5, 0);
-    ERegBoard loxBoard(&Serial8, 1);
+    ERegBoard fuelInjectorBoard(31, 0);
```
Comment (vint-1) [inline src/AC/AC1/EReg.cpp]: EReg ip addresses have changed. See updated ereghat firmware branch

File: src/AC/AC1/EReg.cpp [hunk 11]
```diff
@@ -34,10 +37,12 @@ namespace EReg {
...
+    ERegBoard loxInjectorBoard(32, 1);
+    ERegBoard fuelTankBoard(33, 2);
+    ERegBoard loxTankBoard(34, 3);
```
Comment (vint-1) [inline src/AC/AC1/EReg.cpp]: same

**Review by vint-1 — COMMENTED** [review]
Nice

**Pull request merged**